src/modules/0.0.18/aws_cloudfront/index.ts

```diff
@@ -0,0 +1,288 @@
+import {
+    CloudFront,
+    DistributionConfig,
+    paginateListDistributions,
+    waitUntilDistributionDeployed,
+    DistributionSummary,
+    GetDistributionCommandOutput,
+} from '@aws-sdk/client-cloudfront'
+
+import { AWS, crudBuilder2, paginateBuilder, } from '../../../services/aws_macros'
+import { Distribution, viewerProtocolPolicyEnum, } from './entity'
+import { Context, Crud2, MapperBase, ModuleBase, } from '../../interfaces'
+import { WaiterOptions } from '@aws-sdk/util-waiter'
+
+class DistributionMapper extends MapperBase<Distribution> {
+    module: AwsCloudfrontModule;
+    entity = Distribution;
+    equals = (a: Distribution, b: Distribution) =>
+        Object.is(a.callerReference, b.callerReference) && Object.is(a.comment, b.comment) && Object.is(a.enabled, b.enabled) &&
+            Object.is(a.isIPV6Enabled, b.isIPV6Enabled) && Object.is(a.webACLId, b.webACLId);
```

Comment:
hmm, I do not see here as many comparisons as entity properties. That means that if someone change one of the properties we are not comparing the row could get in a "corrupt" state because it will not have accurate information. If they are not updatable we should still compare them and restore them. This could happen in a follow-up PR, but I think we should leave a `TODO` here.